pkg/operator/controllers/etchosts/machineconfig_controller.go

```diff
@@ -0,0 +1,124 @@
+package etchosts
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"regexp"
+
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/Azure/ARO-RP/pkg/operator"
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers/base"
+	"github.com/Azure/ARO-RP/pkg/util/dynamichelper"
+)
+
+const (
+	EtcHostsMachineConfigControllerName = "DnsmasqMachineConfig"
```

Comment:
I'm not super up-to-speed with the full incident response plan, but I guess this name choice indicates that this controller is intended to completely replace the existing dnsmasq MachineConfig controller?

If so, does it need the additional controllers for watching the ARO cluster CR and the MCPs (like the existing controllers in `pkg/operator/controllers/dnsmasq`)?